funcx_endpoint/tests/stress_tests/test_io_volume.py

```diff
@@ -0,0 +1,83 @@
+import logging
+
+import pytest
+from shared import simple_function
+
+logger = logging.getLogger("test")
+
+
+# @pytest.mark.skip
+@pytest.mark.parametrize("task_count", [10000])
+def test_small_IO_through_s3_executor(fx, endpoint, task_count, result_size=100000):
+    """Launches N tasks that return ~1MB results per task."""
+    futures = {}
+    for _i in range(task_count):
+        kwargs = {"input_data": bytearray(1), "output_size": result_size}
+        future = fx.submit(simple_function, sleep_dur=0, **kwargs, endpoint_id=endpoint)
+        futures[future] = kwargs
+
+    for future in futures:
+        print("Trying future: ", future)
+        x = future.result()
+        expected_result = (
+            len(futures[future]["input_data"]),
+            bytearray(futures[future]["output_size"]),
+        )
+        assert x == expected_result, f"Result does not match, got: {x}"
+
+
+@pytest.mark.parametrize("task_count", [10000])
+def test_IO_through_redis_executor(fx, endpoint, task_count, result_size=10000):
+    """Launches N tasks that return ~10KB results per task that would use redis"""
+    futures = {}
+    for _i in range(task_count):
+        kwargs = {"input_data": bytearray(1), "output_size": result_size}
+        future = fx.submit(simple_function, sleep_dur=0, **kwargs, endpoint_id=endpoint)
+        futures[future] = kwargs
+        # logger.warning(f"Launching futures with {future.task_id}")
+
+    for future in futures:
+        print("Trying future: ", future)
+        try:
+            x = future.result(timeout=600)
+        except TimeoutError:
+            logger.warning(f"Failed to get results for task_id:{future.task_id}")
+            raise
+        expected_result = (
+            len(futures[future]["input_data"]),
+            bytearray(futures[future]["output_size"]),
+        )
+        assert x == expected_result, f"Result does not match, got: {x}"
+
+
+@pytest.mark.skip
+@pytest.mark.parametrize("task_count", [1000])
+# def test_IO_through_S3_executor(fx, endpoint, task_count, result_size=7000000):
+def test_IO_through_S3_executor(fx, endpoint, task_count, result_size=700):
```

Comment:
DId you mean to leave the 7MB version commented out?

Reply:
Fixed.